### PR TITLE
Fix scheduler and dataset shuffling

### PR DESCRIPTION
**Pull request by vwxyzjn**
Closes https://github.com/allenai/open-instruct/issues/537 and fix the dataset shuffling logic.

E.g., https://wandb.ai/ai2-llm/open_instruct_internal/runs/52afsubq/workspace?nw=nwusercostah

image

seems reasonable: we train for 10M episodes, by 1M, the lr decreased from 5e-7 to 4.5e-7.

### CODE REVIEW

File: open_instruct/dpo_tune_cache.py [hunk 1]
```diff
@@ -542,7 +542,7 @@ def main(args: FlatArguments):
             args.max_seq_length,
             args.hf_entity,
         )
-        train_dataset.shuffle(seed=args.seed)
+        train_dataset = train_dataset.shuffle(seed=args.seed)
```
Comment (hamishivi) [inline open_instruct/dpo_tune_cache.py:545]: ah, nice catch

File: open_instruct/grpo_vllm_thread_ray_gtrl.py [hunk 2]
```diff
@@ -642,15 +641,15 @@ def from_pretrained(
         # optim_params = get_optimizer_grouped_parameters(self.policy, weight_decay)
         # self.optimizer = AdamOptimizer(optim_params, lr=args.learning_rate)
         self.optimizer = torch.optim.AdamW(self.policy.parameters(), lr=args.learning_rate)
-        num_training_steps = args.num_training_steps * args.num_train_epochs * args.num_epochs
+        num_scheduler_steps = args.num_training_steps * args.num_epochs * args.num_mini_batches
```
Comment (hamishivi) [inline open_instruct/grpo_vllm_thread_ray_gtrl.py:644]: I guess the scheduler steps every ministep, not ministep*grad acc.

**Review by hamishivi — APPROVED** [review]
Seems good! I didn't run the math on your example but seems valid.

Relatedly, I wonder if we should just set constant LR as the default? Why use the cooldown?

File: open_instruct/grpo_vllm_thread_ray_gtrl.py (hunk 2 above)
Comment (vwxyzjn) [reply open_instruct/grpo_vllm_thread_ray_gtrl.py:644]: This is because we do a gradient step at each mini batches (i.g., local_mini_batch_size)
